app/src/main/java/it/niedermann/owncloud/notes/android/fragment/SearchableBaseNoteFragment.java

```diff
@@ -123,7 +136,36 @@ public boolean onQueryTextChange(String newText) {
                 currentOccurrence = 1;
                 jumpToOccurrence();
                 colorWithText(searchQuery, currentOccurrence);
-                return true;
+            }
+
+            private void queryWithHandler(String newText) {
+                if (delayQueryTask != null) {
+                    delayQueryTask.cancel();
+                    handler.removeCallbacksAndMessages(null);
+                }
+                delayQueryTask = new DelayQueryRunnable(newText);
+                handler.postDelayed(delayQueryTask, delay);
```

Comment:
This should only be delayed if the `newText` length is > 1. Otherwise typing in the first character of the search is also delayed and this feels strange. It should immediately search for the first entered character and only delay all further searches.